test/checking_resource.hpp

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2021 Dmitry Arkhipov (grisumbras@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/json
+//
+
+#ifndef BOOST_JSON_CHECKING_RESOURCE_HPP
+#define BOOST_JSON_CHECKING_RESOURCE_HPP
+
+#include <boost/json/memory_resource.hpp>
+#include <map>
+
+#include "test_suite.hpp"
+
+BOOST_JSON_NS_BEGIN
+
+class checking_resource
+    : public memory_resource
+{
+    memory_resource* base_;
+    std::map<void const*, std::pair<std::size_t, std::size_t>> allocs_;
+
+public:
+    explicit checking_resource(memory_resource& base) noexcept
```

Comment:
The parameter type should be `storage_ptr`. See:
https://github.com/boostorg/json/blob/develop/include/boost/json/monotonic_resource.hpp#L164